packages/storybook-builder-vite/plugins/react-docgen/index.js

```diff
@@ -0,0 +1,59 @@
+const ReactDocgen = require('react-docgen');
+const { promises: fs } = require('fs');
+const path = require('path');
+const actualNameHandler = require('./actualNameHandler');
+const defaultHandlers = Object.values(ReactDocgen.handlers).map(handler => handler);
+const resolver = ReactDocgen.resolver.findAllExportedComponentDefinitions;
+
+/**
+ * This is a cut-down version of https://github.com/storybookjs/babel-plugin-react-docgen/blob/2df98996a51d7f537c29b63b93cc6b7687e2baaa/src/index.js#L24
+ * This doesn't aim to be generic or allow customization so a lot of code was removed.
+ */
+
+module.exports = function () {
+    return {
+        name: 'react-docgen',
+
+        async transform(src, id) {
+            if (/\.(mjs|tsx|jsx)$/.test(id)) {
+                let docgenResults = [];
+                try {
+                    const code = await fs.readFile(id, 'utf-8');
+                    const handlers = [...defaultHandlers, actualNameHandler];
+                    docgenResults = ReactDocgen.parse(code, resolver, handlers, {
+                        filename: id,
+                    });
+
+                    if (docgenResults && !Array.isArray(docgenResults)) {
+                        docgenResults = [docgenResults];
+                    }
+                } catch (e) {
+                    // this is for debugging the error only, do not ship this console log or else it pollutes the webpack output
+                    // console.log(id, e);
+                    return;
+                }
+
+                let docGenExtra = '';
+
+                docgenResults.forEach((docGenResult) => {
+                    const actualName = docGenResult.actualName;
+                    //  Escape windows paths
+                    const filePath = path.relative(process.cwd(), id).replace(/\\/g, '\\\\');
+                   docGenExtra += `
+                   ${actualName}.__docgenInfo = ${JSON.stringify(docGenResult, null, 2)};
```

Comment:
I basically reverse engineered this from the babel injection code (https://github.com/storybookjs/babel-plugin-react-docgen/blob/2df98996a51d7f537c29b63b93cc6b7687e2baaa/src/index.js#L102)